### ISSUE TITLE
999999999^9999999^999^99^999^99999 returns an error

### ISSUE THREAD

**Issue by Tpt** [ProjetPP/PPP-CAS#17]
Not sure of what we should do on such big inputs

**Comment by s-i-newton** [comment]
Not my fault! Sympy tries to do the computation since it knows it will 
succeed but it is killed by timeout of the subprocess. There is not way 
to predict if a computation will be long. If you disagree, think about 
the halting problem. So, we cannot do anything. I think the best is just 
to consider this input as "I cannot handle it", just as when the input 
is not a mathematical formula and to return... nothing.

Le 02/02/2015 17:20, Thomas Tanon a écrit :
>
> Not sure of what we should do on such big inputs
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/ProjetPP/PPP-CAS/issues/17>.
>

-- 

Marc Chevalier

ENS de Lyon
Site Monod
M1 Informatique Fondamentale

**Comment by Tpt** [comment]
Yes, I agree that it is difficult. I've opened this issue thinking more "we should track this issue" than "we should solve this issue now".

**Comment by s-i-newton** [comment]
I think there is nothing to do. Or add a error message saying "Oups! Too 
long computation, sorry. Do it yourself or buy Mathematica" But the 
doesn't change really anything.

**Comment by eddycaron** [comment]
True. You should have at least a timeout to decide when you cancel the computation. Just one way to avoid malicious guys who try to stress platypus for bad reason.

---------------------------------------
Eddy Caron. Mcf HDR ENS Lyon 
ENS Lyon - LIP - Projet AVALON 
E-Mail : Eddy.Caron@ens-lyon.fr
[http://graal.ens-lyon.fr/~ecaron]
---------------------------------------

**Comment by s-i-newton** [comment]
We have a timeout for a long time, almost before the first deployement.
https://github.com/ProjetPP/PPP-CAS/blob/master/ppp_cas/evaluator.py#L25

**Comment by s-i-newton** [comment]
Moreover, I'm pretty sure the timeout works because, since we use 
platypus-angel which is less powerful as the previous server, the 
request "Integrate[Sin[x*y], {x, 0, 1}, {y, 0, x}]" (one of my longer 
test) caused an error until we increase the timeout. Since, that works 
again correctly.
